### PR TITLE
HSD8-549 Allow users to choose which embed player to use

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Allow users to choose which embed player to use

# Needed By (Date)
- Someday

# Urgency
- Low

# Steps to Test
1. Embed a soundcloud audio
1. ensure "Classic Embed" and "Visual Embed" radio button are visible and function as expected.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: src/Plugin/MediaEmbedDialog/SoundCloud.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+class SoundCloud extends MediaEmbedDialogBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function isApplicable() {
+    if ($this->entity instanceof MediaInterface && $this->entity->bundle() == 'audio') {
+      $source_field = static::getMediaSourceField($this->entity);
+      $field_value = $this->entity->get($source_field)->getString();
+      return strpos($field_value, 'soundcloud') !== FALSE;
```
Comment (sherakama) [inline src/Plugin/MediaEmbedDialog/SoundCloud.php]: I suppose it is unlikely but would it be possible that some other audio source has soundclound in the value? Eg: A url to itunes with a song titled soundcloud? Perhaps this should be a bit stronger?

File: src/Plugin/audio_embed_field/Provider/StanfordSoundCloud.php [hunk 2]
```diff
@@ -57,6 +56,21 @@ public static function getIdFromInput($input) {
...
+  /**
+   * {@inheritdoc}
+   */
+  public function renderEmbedCode($width, $height, $autoplay) {
+    $render = parent::renderEmbedCode($width, $height, $autoplay);
+    $video_data = static::getVideoData($this->input);
+
+    // Fix the url if the input is actually a playlist instead of a track.
+    if (strpos($video_data['html'], 'playlist') !== FALSE) {
+      $render['#url'] = str_replace('/tracks/', '/playlists/', $render['#url']);
```
Comment (sherakama) [inline src/Plugin/audio_embed_field/Provider/StanfordSoundCloud.php]: That seems like a nice helper, but does it give the user the desired outcome just by swapping out the part in the url? Does the right track play? Could that be confusing?

File: src/Plugin/MediaEmbedDialog/File.php [hunk 3]
```diff
@@ -49,7 +49,6 @@ public function alterDialogForm(array &$form, FormStateInterface $form_state) {
   public function preRender(array $element) {
     $source_field = static::getMediaSourceField($element['#media']);
     $element[$source_field][0]['#description'] = $element['#display_settings']['description'];
-    $element['#cache']['max-age'] = 0;
```
Comment (sherakama) [inline src/Plugin/MediaEmbedDialog/File.php]: Why the change in heart?

File: src/Plugin/MediaEmbedDialog/SoundCloud.php (hunk 1 above)
Comment (pookmish) [reply src/Plugin/MediaEmbedDialog/SoundCloud.php]: Should be better now

File: src/Plugin/audio_embed_field/Provider/StanfordSoundCloud.php (hunk 2 above)
Comment (sherakama) [reply src/Plugin/audio_embed_field/Provider/StanfordSoundCloud.php]: neat.
